### DISCUSSION TITLE
Unable to have 'no strict hostname' and 'backchannel dynamic feature' activated both at the same time with hostname:v2

### DISCUSSION THREAD

**Discussion by kakuzei** [keycloak/keycloak#31983] [Q&A] [answered]
Hello, 
I am trying to upgrade Keycloak from version 24 to version 25 and I'm having an issue with the `hostname:v2` feature.
It looks like there is no configuration suitable for my use case as I would no strict hostname and the backchannel dynamic feature activated both at the same time.

I am running one instance of Keycloak in production that is available using multiple URL (for example: https://website1.com and https://website2.com).
I am relying on code flow with backend to authenticate users.
I configured everything to have the users of https://website1.com using Keycloak on https://website1.com/auth and the users of https://website2.com using the same instance of Keycloak on https://website2.com/auth.
I'm using the same realm and the same client_id for both websites (in fact both websites are targeting the same instance of the same application). 
Redirect URI are configured for both websites.

I wanted to start Keycloak with the following configuration:
```
KC_HOSTNAME=
KC_HOSTNAME_BACKCHANNEL_DYNAMIC=true
KC_HOSTNAME_STRICT=false
```
But a hostname must be provided when `KC_HOSTNAME_BACKCHANNEL_DYNAMIC` is set to `true`.

So, I configured Keycloak like this:
```
KC_HOSTNAME=https://website1.com/auth
KC_HOSTNAME_BACKCHANNEL_DYNAMIC=true
KC_HOSTNAME_STRICT=false
```

With this configuration the code flow authentication works well using https://website1.com.
But it failed with https://website2.com as Keycloak redirect me to https://website1.com/auth/realms/abc/login-actions/authenticate?session_code=xxx&execution=xyz&client_id=my-client&tab_id=xxx&client_data=abc during the code flow authentication.

It works well when I'm using my old configuration with the `hostname:v1` feature.
Am I missing something here? 
Is there a way to configure Keycloak 25 with `hostname:v2` feature to support the behavior I'm having with the hostname:v1` feature?

Thank you to have read this :)

**Comment by darius-m** [comment, marked as answer]
I think the backchannel dynamic configuration may not be required. Reading https://www.keycloak.org/server/hostname, I think you don't actually need to set it to `true`, since it only affects if the backend URL must strictly match the frontend URL (which would be dynamically defined when `hostname-strict` is set to `false`).

**Comment by kakuzei** [comment]
@darius-m thank you for your answer.

As you said, I can leave KC_HOSTNAME_BACKCHANNEL_DYNAMIC to false and rely only on KC_HOSTNAME_STRICT.

I confirm I am now able to use the hostname:v2 feature in production for my specific use case using.
For the record, I am using the following configuration:
```
KC_HOSTNAME=
KC_HOSTNAME_BACKCHANNEL_DYNAMIC=false
KC_HOSTNAME_STRICT=false
KC_PROXY_HEADERS=xforwarded
```
